base/broadcast.jl

```diff
@@ -1261,4 +1261,10 @@ end
 end
 @inline broadcasted(::S, f, args...) where S<:BroadcastStyle = Broadcasted{S}(f, args)
 
+struct BroadcastOp{F}
+    f::F
+end
+
+(op::BroadcastOp)(x...) = op.f.(x...)
```

Comment:
Have you checked that the compiler specializes on F here? i.e. that performance is unchanged?

Reply:
Good catch! There seems to be room for improvement:
```
julia> using BenchmarkTools

julia> a = [rand(10, 10) for _ in 1:100];

julia> b = [rand(10, 10) for _ in 1:100];

julia> @btime map((x, y) -> (Base.@_inline_meta; x .* y), $a, $b);
  14.528 μs (101 allocations: 88.38 KiB)

julia> @btime map(.*, $a, $b);
  53.660 μs (401 allocations: 97.75 KiB)
```
Shouldn't be too difficult to fix that.

Reply:
Adding `@inline` here fixes this performance differences for me.

Reply:
Maybe you just need `(op::BroadcastOp{F})(x...) where {F} = op.f.(x...)`?  Adding an explicit type parameter usually tells the compiler to specialize on function types.

Reply:
I have tried that as well, but it didn't make any difference. It seems like the compiler needs to be explicitely told to inline here.